MediaGalleryUi/Controller/Adminhtml/Image/Delete.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Controller\Adminhtml\Image;
+
+use Exception;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Cms\Model\Wysiwyg\Images\Storage;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\MediaGalleryApi\Model\Asset\Command\GetByIdInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Controller deleting the media gallery content
+ */
+class Delete extends Action
+{
+    private const HTTP_OK = 200;
+    private const HTTP_INTERNAL_ERROR = 500;
+    private const HTTP_BAD_REQUEST = 400;
+
+    /**
+     * @see _isAllowed()
+     */
+    public const ADMIN_RESOURCE = 'Magento_Cms::media_gallery';
+
+    /**
+     * @var GetByIdInterface
+     */
+    private $getAssetById;
+
+    /**
+     * @var Storage
+     */
+    private $imagesStorage;
+
+
+    /**
+     * @var Storage
+     */
+    private $filesystem;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Delete constructor.
+     *
+     * @param Context $context
+     * @param GetByIdInterface $getAssetById
+     * @param Storage $imagesStorage
+     * @param Filesystem $filesystem
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        Context $context,
+        GetByIdInterface $getAssetById,
+        Storage $imagesStorage,
+        Filesystem $filesystem,
+        LoggerInterface $logger
+    ) {
+        parent::__construct($context);
+
+        $this->getAssetById = $getAssetById;
+        $this->imagesStorage = $imagesStorage;
+        $this->filesystem = $filesystem;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute()
+    {
+        /** @var Json $resultJson */
+        $resultJson = $this->resultFactory->create(ResultFactory::TYPE_JSON);
+        $imageId = (int) $this->getRequest()->getParam('image_id');
+
+        if (!$imageId) {
+            $responseContent = [
+                'success' => false,
+                'message' => __('Image ID is required.'),
+            ];
+            $resultJson->setHttpResponseCode(self::HTTP_BAD_REQUEST);
+            $resultJson->setData($responseContent);
+
+            return $resultJson;
+        }
+
+        try {
+            $image = $this->getAssetById->execute($imageId);
+            $mediaFilePath = $image->getPath();
+            $mediaDirectory = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA);
+            $absoluteMediaPath = $mediaDirectory->getAbsolutePath();
+            $this->imagesStorage->deleteFile($absoluteMediaPath . $mediaFilePath);
```

Comment:
Let's make sure that `$mediaFilePath` is relative to media directory by 
```suggestion
            $this->imagesStorage->deleteFile($mediaDirectory->getAbsolutePath($mediaFilePath));
```

Reply:
Hi @sivaschenko
Thank you for this suggestion.
I've tried to add this modification but I faced an issue.
The `getAbsolutePath()` method removes doubled slashes (`//`) and it leads to an issue with the wrong relative file path in this place: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/MediaGallery/Plugin/Product/Gallery/Processor.php#L68.
The slash (/) from the start of the relative path is removed and this relative path can not be matched with an asset here: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/MediaGallery/Model/Asset/Command/DeleteByPath.php#L63 
I propose to leave the concatenation of absolute media path and asset image path (`$absoluteMediaPath . $mediaFilePath`).
Also, we added the checking for an asset file existing in the media. (https://github.com/magento/adobe-stock-integration/pull/970/commits/db7b00b789c2953b3d37cfd9c065aa5078b433d1#diff-5c84bd8e50cdb5894cbeba40eb73163eR69-R71)
@sivaschenko please, let me know your thoughts regarding this.
Thank you!